ext/src/ruby_api/mod.rs

```diff
@@ -28,8 +28,27 @@ pub fn root() -> RModule {
     *memoize!(RModule: define_module("Wasmtime").unwrap())
 }
 
+// This Struct is a placeholder for documentation, so that we can hang methods
+// to it and have yard-rustdoc discover them.
+/// @yard
+/// @module
+pub struct Wasmtime();
+impl Wasmtime {
```

Comment:
Did you consider adding this under a different namespace? Personally I find it confusing to have this under the Wasmtime namespace, mainly because Wasmtime doesn't offer this functionality.

Reply:
Do you mean on the Ruby layer, this struct specifically? Personally it doesn't bother me as it'll end up on the `Wasmtime` constant. Alternatively, we could name this `Root`, but that feels worse.

> Personally I find it confusing to have this under the Wasmtime namespace, mainly because Wasmtime doesn't offer this functionality.

I see your point, it's in a different crate. From the bindings perspective though, it's often bundled with Wasmtime itself:

- Go has [`wasmtime.Wat2Wasm`](https://github.com/bytecodealliance/wasmtime-go/blob/4b3a40c7a33dca1cfa4e06e6f40e06816f7d2421/README.md?plain=1#L75)
- C has [`wasmtime_wat2wasm`](https://github.com/bytecodealliance/wasmtime/blob/d54a27d0ea83090bf1bd46032a2129677f70df0f/crates/c-api/src/wat2wasm.rs)
- .NET has [`Wasmtime.Module.ConvertText`](https://github.com/bytecodealliance/wasmtime-dotnet/blob/992efb0baee78bad1b4f9cc0a8a5b363313a63c9/src/Module.cs#L326)

Reply:
.NET approach seems to be the cleanest to me, but since there's a precedent on this I'll leave it to you if you want to change it.

Reply:
Maybe put it in `Wasmtime::Utils`